tools/clang/lib/CodeGen/CGHLSLMS.cpp

```diff
@@ -4246,6 +4247,70 @@ void CGMSHLSLRuntime::SetPatchConstantFunctionWithAttr(
   
 }
 
+// Returns true a global value is being updated
+static bool IsWriteEnabledGlobalRec(Value *V, std::set<Value*> &visited) {
+  bool isWriteEnabled = false;
+  if (V && visited.find(V) == visited.end()) {
+    visited.insert(V);
+    for (User *U : V->users()) {
+      if (isa<StoreInst>(U))
+        isWriteEnabled = true;
+      else if (GetElementPtrInst *GEP = dyn_cast<GetElementPtrInst>(U)) {
+        for(User *GU : GEP->users())
+          if(isa<StoreInst>(GU))
+            isWriteEnabled = true;
+          else
+            isWriteEnabled |= IsWriteEnabledGlobalRec(GU, visited);
+      } else isWriteEnabled |= IsWriteEnabledGlobalRec(U, visited);
```

Comment:
You should break as soon as isWriteEnabled == true, since you don't need to search anymore.